src/SignalR/clients/java/signalr/core/src/main/java/com/microsoft/signalr/CallbackMap.java

```diff
@@ -55,4 +60,16 @@ public void remove(String key) {
             lock.unlock();
         }
     }
+
+    public void remove(String key, InvocationHandler handler) {
+        try {
+            lock.lock();
+            List<InvocationHandler> handlers = this.handlers.get(key);
+            if (handlers != null) {
+                handlers.remove(handler);
```

Comment:
Instead of mutating the existing list, wouldn't it likely be more efficient to make this copy-on-write and replace the reference so we don't have to copy it every time in get()?

Reply:
Hmm, that's a good point. Looks like that's what we do in the .NET client so at least we were smart there 😄